Makefile

```diff
@@ -5,6 +5,39 @@ DOCKER_BUF := $(DOCKER) run --rm -v $(CURDIR):/workspace --workdir /workspace bu
 IMAGE := ghcr.io/tendermint/docker-build-proto:latest
 DOCKER_PROTO_BUILDER := docker run -v $(shell pwd):/workspace --workdir /workspace $(IMAGE)
 PROJECTNAME=$(shell basename "$(PWD)")
+LEDGER_ENABLED ?= true
+
+
+# process build tags
+build_tags =
+ifeq ($(LEDGER_ENABLED),true)
+  ifeq ($(OS),Windows_NT)
+    GCCEXE = $(shell where gcc.exe 2> NUL)
+    ifeq ($(GCCEXE),)
+      $(error gcc.exe not installed for ledger support, please install or set LEDGER_ENABLED=false)
+    else
+      build_tags += ledger
+    endif
+  else
+    UNAME_S = $(shell uname -s)
+    ifeq ($(UNAME_S),OpenBSD)
+      $(warning OpenBSD detected, disabling ledger support (https://github.com/cosmos/cosmos-sdk/issues/1988))
+    else
+      GCC = $(shell command -v gcc 2> /dev/null)
+      ifeq ($(GCC),)
+        $(error gcc not installed for ledger support, please install or set LEDGER_ENABLED=false)
+      else
+        build_tags += ledger
+      endif
+    endif
+  endif
+endif
+build_tags := $(strip $(build_tags))
+whitespace :=
+whitespace := $(whitespace) $(whitespace)
```

Comment:
I don't quite understand this. Does this create three whitespaces?

Reply:
It was copied from [here](https://github.com/osmosis-labs/osmosis/blob/c26756d897641ee441070986d898e72f94f47807/Makefile#L53-L54) but I think the original variable name `whitespace` is misleading. It's actually empty.

Ref: https://groups.google.com/g/gnu.utils.help/c/Kph-2Mk4NxU

```
build_tags += test
build_tags := $(strip $(build_tags))

empty :=
whitespace := $(empty) $(empty)
comma := ,

# convert build_tags from a whitespace seperated list to a comma seperated list
build_tags_comma_sep := $(subst $(whitespace),$(comma),$(build_tags))

debug:
	@echo $(build_tags)
	@echo $(build_tags_comma_sep)
.PHONY: debug
```

Testing

```
make debug
ledger test
ledger,test
```

Reply:
Resolved via: https://github.com/celestiaorg/celestia-app/pull/2014/commits/c80fe5d217fa372c17e319d6862b6ca0e527550d

To answer your original question

> Does this create three whitespaces?

I think it just creates a singular whitespace character in between the two `$(empty)`s

Reply:
> I think it creates a singular whitespace character in between the two $(empty)s

lol, but why?

Reply:
I briefly explored alternative ways to replace whitespace with commas that are more readable. My attempts _didn't_ actually replace whitespace with commas correctly so going to keep as-is. Happy to replace this with something more readable in a follow-up PR if anyone has suggestions.